### PR TITLE
[Docs] Refactor and add additional API docs

### LINKED ISSUE #2710: [Docs] Core Python API docs are incomplete
Our [Python API docs](https://skypilot.readthedocs.io/en/latest/reference/api.html) do not include the python equivalent methods for:
* View job queue (`sky queue`)
* Cancel task (`sky cancel`)
* Get logs (`sky logs`)

### PR DESCRIPTION
**Pull request by Michaelvll**
~~Refactor the `launch`, `exec`, `spot_launch` to `core.py`, so as to make it easier to provide Restful API.~~ Seems we don't need to refactor it as we should create another API file anyway.

Also, we add docs for `sky.queue`, etc to fix #2710

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] Rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/core.py [hunk 1]
```diff
@@ -906,13 +938,19 @@ def spot_cancel(name: Optional[str] = None,
...
+                   job_id: Optional[int] = None,
+                   follow: bool = True) -> None:
     # NOTE(dev): Keep the docstring consistent between the Python API and CLI.
     """Tail logs of managed spot jobs.
 
     Please refer to the sky.cli.spot_logs for the document.
 
+    Args:
+        name: name of the job to tail.
+        job_id: job id to tail.
```
Comment (concretevitamin) [inline sky/core.py:951]:
```suggestion
        job_id: job id to tail. If None, ...
```

File: sky/core.py [hunk 2]
```diff
@@ -703,8 +722,9 @@ def job_status(cluster_name: str,
     Args:
         cluster_name: (str) name of the cluster.
         job_ids: (List[str]) job ids. If None, get the status of the last job.
+        stream_logs: (bool) whether to stream logs.
     Returns:
-        Dict[Optional[int], Optional[job_lib.JobStatus]]: A mapping of job_id to
+        Dict[Optional[int], Optional[sky.JobStatus]]: A mapping of job_id to
         job statuses. The status will be None if the job does not exist.
         If job_ids is None and there is no job on the cluster, it will return
         {None: None}.
```
Comment (concretevitamin) [inline sky/core.py:730]: Our manually annotated `(type)` can be removed

File: docs/source/reference/api.rst [hunk 3]
```diff
@@ -52,6 +52,62 @@ sky.autostop
...
+sky.tail_logs
+~~~~~~~~~~~~~~~~~
+
+.. autofunction:: sky.tail_logs
+
+
+sky.download_logs
+~~~~~~~~~~~~~~~~~~~~~
+
+.. autofunction:: sky.download_logs
```
Comment (concretevitamin) [inline docs/source/reference/api.rst:79]: Our manually annotated `(type)` can be removed

File: docs/source/reference/api.rst [hunk 4]
```diff
@@ -52,6 +52,62 @@ sky.autostop
...
+
+.. autofunction:: sky.spot_cancel
+
+
+sky.spot_tail_logs
+~~~~~~~~~~~~~~~~~~
+
+.. autofunction:: sky.spot_tail_logs
+
 .. _sky-dag-ref:
```
Comment (concretevitamin) [inline docs/source/reference/api.rst:112]: Probably should expose the referenced `sky.JobStatus` enum below Task and Resources?

**Review by concretevitamin — COMMENTED** [review]
Awesome @Michaelvll! Quick comments.

File: docs/source/reference/api.rst (hunk 3 above)
Comment (Michaelvll) [reply docs/source/reference/api.rst:79]: Removed. Thanks!

File: docs/source/reference/api.rst (hunk 4 above)
Comment (Michaelvll) [reply docs/source/reference/api.rst:112]: Added. Thanks!

File: sky/core.py (hunk 2 above)
Comment (Michaelvll) [reply sky/core.py]: Removed. Thanks!

File: sky/core.py (hunk 1 above)
Comment (Michaelvll) [reply sky/core.py]: Added. Thanks!

**Pull request closed without merging**